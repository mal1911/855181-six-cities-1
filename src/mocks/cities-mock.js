export const citiesMock = [
  {
    id: 1,
    name: `Paris`,
  },
  {
    id: 2,
    name: `Cologne`,
  },
  {
    id: 3,
    name: `Brussels`,
  },
  {
    id: 4,
    name: `Amsterdam`,
  },
  {
    id: 5,
    name: `Hamburg`,
  },
  {
    id: 6,
    name: `Dusseldorf`,
  },
];

