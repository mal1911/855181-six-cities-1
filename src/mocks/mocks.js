export const offersData = [
  {
    id: 1,
    isPremium: true,
    city: `Paris`,
    title: `Paris Beautiful & luxurious apartment at great location`,
    price: 10,
    rating: 1,
    type: `Apartment`,
    imgSrc: `img/apartment-01.jpg`,
    coordinates: [52.3909553943508, 4.85309666406198],
  },
  {
    id: 2,
    isPremium: false,
    city: `Paris`,
    title: `Paris Wood and stone place`,
    price: 20,
    rating: 2,
    type: `Apartment`,
    imgSrc: `img/apartment-02.jpg`,
    coordinates: [52.369553943508, 4.85309666406198],
  },
  {
    id: 3,
    isPremium: false,
    city: `Paris`,
    title: `Paris Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 4,
    isPremium: false,
    city: `Paris`,
    title: `Paris Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 5,
    isPremium: true,
    city: `Cologne`,
    title: `Cologne Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 6,
    isPremium: false,
    city: `Cologne`,
    title: `Cologne Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 7,
    isPremium: true,
    city: `Brussels`,
    title: `Brussels Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 8,
    isPremium: false,
    city: `Brussels`,
    title: `Brussels Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 9,
    isPremium: true,
    city: `Amsterdam`,
    title: `Amsterdam Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 10,
    isPremium: false,
    city: `Amsterdam`,
    title: `Amsterdam Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 11,
    isPremium: true,
    city: `Hamburg`,
    title: `Hamburg Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 12,
    isPremium: false,
    city: `Hamburg`,
    title: `Hamburg Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 13,
    isPremium: true,
    city: `Dusseldorf`,
    title: `Dusseldorf Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 14,
    isPremium: false,
    city: `Dusseldorf`,
    title: `Dusseldorf Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
];

export const citiesData = [
  {
    name: `Paris`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Cologne`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Brussels`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Amsterdam`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Hamburg`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Dusseldorf`,
    coordinates: [52.38333, 4.9],
  },
];
