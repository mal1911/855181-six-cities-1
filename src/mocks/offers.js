export const offers = [
  {
    title: `Beautiful & luxurious apartment at great location`,
    price: 10,
    rating: 1,
    type: `Apartment`,
    imgSrc: `img/apartment-01.jpg`
  },
  {
    title: `Wood and stone place`,
    price: 20,
    rating: 2,
    type: `Apartment`,
    imgSrc: `img/apartment-02.jpg`
  },
  {
    title: `Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`
  },
  {
    title: `Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`
  },
];
