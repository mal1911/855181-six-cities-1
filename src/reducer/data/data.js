import {transformOfferForLoading, transformCommentsForLoading} from "../../transform-data";
import {HTML_STATUS} from "../../constants";

const initialState = {
  activeCityIndex: 0,
  activeOrderIndex: 0,
  activeOfferId: 0,
  offersData: [],
  favoritesData: [],
  commentsData: [],
  offersError: null,
  favoritesError: null,
  commentsError: null,
  isLoading: true,
};

const ActionType = {
  CHANGE_LOAD_STATUS: `CHANGE_LOAD_STATUS`,
  CHANGE_OFFERS_DATA: `CHANGE_OFFERS_DATA`,
  CHANGE_FAVORITES_DATA: `CHANGE_FAVORITES_DATA`,
  CHANGE_COMMENTS_DATA: `CHANGE_COMMENTS_DATA`,
  CHANGE_ACTIVE_CITY_INDEX: `CHANGE_ACTIVE_CITY_INDEX`,
  CHANGE_ACTIVE_ORDER_INDEX: `CHANGE_ACTIVE_ORDER_INDEX`,
  CHANGE_ACTIVE_OFFER_ID: `CHANGE_ACTIVE_OFFER_ID`,
  CHANGE_OFFERS_ERROR_STATUS: `CHANGE_OFFERS_ERROR_STATUS`,
  CHANGE_FAVORITES_ERROR_STATUS: `CHANGE_FAVORITES_ERROR_STATUS`,
  CHANGE_COMMENTS_ERROR_STATUS: `CHANGE_COMMENTS_ERROR_STATUS`,
};

const ActionCreator = {
  changeLoadStatus: (status) => ({
    type: ActionType.CHANGE_LOAD_STATUS,
    payload: status,
  }),
  changeOffersData: (data) => {
    return {
      type: ActionType.CHANGE_OFFERS_DATA,
      payload: data,
    };
  },
  changeFavoritesData: (data) => {
    return {
      type: ActionType.CHANGE_FAVORITES_DATA,
      payload: data,
    };
  },
  changeCommentsData: (data) => {
    return {
      type: ActionType.CHANGE_COMMENTS_DATA,
      payload: data,
    };
  },
  changeActiveCityIndex: (index) => ({
    type: ActionType.CHANGE_ACTIVE_CITY_INDEX,
    payload: index,
  }),
  changeActiveOrderIndex: (index) => ({
    type: ActionType.CHANGE_ACTIVE_ORDER_INDEX,
    payload: index,
  }),
  changeActiveOfferId: (index) => ({
    type: ActionType.CHANGE_ACTIVE_OFFER_ID,
    payload: index
  }),
  changeOffersErrorStatus: (error) => ({
    type: ActionType.CHANGE_OFFERS_ERROR_STATUS,
    payload: error,
  }),
  changeFavoritesErrorStatus: (error) => ({
    type: ActionType.CHANGE_FAVORITES_ERROR_STATUS,
    payload: error,
  }),
  changeCommentsErrorStatus: (error) => ({
    type: ActionType.CHANGE_COMMENTS_ERROR_STATUS,
    payload: error,
  }),
};


const beforeCommentsLoading = (dispatch) => {
  dispatch(ActionCreator.changeLoadStatus(true));
  dispatch(ActionCreator.changeCommentsErrorStatus(null));
};

const successCommentsLoading = (dispatch, data) => {
  dispatch(ActionCreator.changeCommentsData(
      data.map((obj) => transformCommentsForLoading(obj)))
  );
  dispatch(ActionCreator.changeLoadStatus(false));
};

const errorCommentsLoading = (dispatch, err) => {
  dispatch(ActionCreator.changeCommentsErrorStatus(err));
  dispatch(ActionCreator.changeLoadStatus(false));
};

const beforeOffersLoading = (dispatch) => {
  dispatch(ActionCreator.changeLoadStatus(true));
  dispatch(ActionCreator.changeOffersErrorStatus(null));
  dispatch(ActionCreator.changeFavoritesErrorStatus(null));
};


const Operation = {
  loadOffersData: () => (dispatch, _getState, api) => {
    beforeOffersLoading(dispatch);
    return api.get(`/hotels`)
      .then((response) => {
        const data = response.data.map((obj) => transformOfferForLoading(obj));
        dispatch(ActionCreator.changeOffersData(data));
        dispatch(ActionCreator.changeLoadStatus(false));
      })
      .catch((err) => {
        dispatch(ActionCreator.changeOffersErrorStatus(err));
        dispatch(ActionCreator.changeLoadStatus(false));
      });
  },

  loadFavoritesData: (onGoLogin) => (dispatch, _getState, api) => {
    dispatch(ActionCreator.changeLoadStatus(true));
    dispatch(ActionCreator.changeFavoritesErrorStatus(null));

    return api.get(`/favorite`)
      .then((response) => {
        const data = response.data.map((obj) => transformOfferForLoading(obj));
        dispatch(ActionCreator.changeFavoritesData(data));
        dispatch(ActionCreator.changeLoadStatus(false));
      })
      .catch((err) => {
        if (err.status === HTML_STATUS.FORBIDDEN) {
          onGoLogin();
        } else {
          dispatch(ActionCreator.changeFavoritesErrorStatus(err));
          dispatch(ActionCreator.changeLoadStatus(false));
        }
      });
  },

  changeFavoriteStatus: (id, status, offersData, favoritesData, onGoLogin) => (dispatch, _getState, api) => {
    beforeOffersLoading(dispatch);
    return api.post(`/favorite/${id}/${status}`)
      .then((response) => {
        const newObj = transformOfferForLoading(response.data);
        let newData = offersData.map(
            (obj) => newObj.id === obj.id ? Object.assign({}, newObj) : obj);
        dispatch(ActionCreator.changeOffersData(newData));
        const index = favoritesData.findIndex((obj) => obj.id === id);
        newData = [...favoritesData.slice(0, index), ...favoritesData.slice(index + 1)];
        dispatch(ActionCreator.changeFavoritesData(newData));
        dispatch(ActionCreator.changeLoadStatus(false));
      })
      .catch((err) => {
        if (err.status === HTML_STATUS.FORBIDDEN) {
          onGoLogin();
        } else {
          dispatch(ActionCreator.changeFavoritesErrorStatus(err));
        }
        dispatch(ActionCreator.changeLoadStatus(false));
      });
  },

  loadCommentsData: (id) => (dispatch, _getState, api) => {
    beforeCommentsLoading(dispatch);
    return api.get(`/comments/${id}`)
      .then((response) => {
        successCommentsLoading(dispatch, response.data);
      })
      .catch((err) => {
        errorCommentsLoading(dispatch, err);
      });
  },

  saveCommentObj: (id, commentObj) => (dispatch, _getState, api) => {
    beforeCommentsLoading(dispatch);
    return api.post(`/comments/${id}`, commentObj)
      .then((response) => {
        successCommentsLoading(dispatch, response.data);
      })
      .catch((err) => {
        errorCommentsLoading(dispatch, err);
      });
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_LOAD_STATUS:
      return Object.assign({}, state, {
        isLoading: action.payload
      });
    case ActionType.CHANGE_OFFERS_DATA: {
      return Object.assign({}, state, {
        offersData: action.payload,
      });
    }
    case ActionType.CHANGE_FAVORITES_DATA:
      return Object.assign({}, state, {
        favoritesData: action.payload,
      });
    case ActionType.CHANGE_COMMENTS_DATA:
      return Object.assign({}, state, {
        commentsData: action.payload,
      });
    case ActionType.CHANGE_ACTIVE_CITY_INDEX:
      return Object.assign({}, state, {
        activeCityIndex: action.payload,
      });
    case ActionType.CHANGE_ACTIVE_ORDER_INDEX:
      return Object.assign({}, state, {
        activeOrderIndex: action.payload
      });
    case ActionType.CHANGE_ACTIVE_OFFER_ID: {
      return Object.assign({}, state, {
        activeOfferId: action.payload,
      });
    }
    case ActionType.CHANGE_OFFERS_ERROR_STATUS:
      return Object.assign({}, state, {
        offersError: action.payload
      });
    case ActionType.CHANGE_FAVORITES_ERROR_STATUS:
      return Object.assign({}, state, {
        favoritesError: action.payload
      });
    case ActionType.CHANGE_COMMENTS_ERROR_STATUS:
      return Object.assign({}, state, {
        commentsError: action.payload
      });
  }
  return state;
};

export {
  ActionCreator,
  ActionType,
  Operation,
  reducer,
};
