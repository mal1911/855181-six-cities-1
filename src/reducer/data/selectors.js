import {createSelector} from "reselect";
import NameSpace from "../name-spaces";
import {MAX_CITIES} from "../../constants";

const NAME_SPACE = NameSpace.DATA;

const SORT_FUNCS = [
  null,
  (a, b) => a.price < b.price,
  (a, b) => a.price > b.price,
  (a, b) => a.rating < b.rating,
];

export const getOffersData = (state) => {
  return state[NAME_SPACE].offersData;
};

export const getActiveCityIndex = (state) => {
  return state[NAME_SPACE].activeCityIndex;
};

export const getActiveOrderIndex = (state) => {
  return state[NAME_SPACE].activeOrderIndex;
};

export const getActiveOfferId = (state) => {
  return state[NAME_SPACE].activeOfferId;
};

export const getLoadStatus = (state) => {
  return state[NAME_SPACE].isLoading;
};

export const getOffersError = (state) => {
  return state[NAME_SPACE].offersError;
};

export const getFavoritesData = (state) => {
  return state[NAME_SPACE].favoritesData;
};

export const getFavoritesError = (state) => {
  return state[NAME_SPACE].favoritesError;
};

export const getCommentsData = (state) => {
  return state[NAME_SPACE].commentsData.sort((a, b) => a.date < b.date);
};

export const getCommentsError = (state) => {
  return state[NAME_SPACE].commentsError;
};

export const getCitiesData = createSelector(
    getOffersData,
    (offersData) => {
      const citiesData = [];
      if (offersData.length) {
        offersData.forEach((offerObj) => {
          if (!citiesData.find((cityObj) => cityObj.name === offerObj.city.name)) {
            citiesData.push(offerObj.city);
          }
        });
      }
      return citiesData.slice(0, MAX_CITIES);
    }
);

export const getResultOffersData = createSelector(
    getOffersData,
    getCitiesData,
    getActiveCityIndex,
    getActiveOrderIndex,
    (offersData, citiesData, activeCityIndex, activeOrderIndex) => {
      let resultOffersData = [];
      if (offersData.length && citiesData.length) {
        resultOffersData = offersData.filter((offerObj) => offerObj.city.name === citiesData[activeCityIndex].name);
        if (activeOrderIndex) {
          resultOffersData = resultOffersData.sort(SORT_FUNCS[activeOrderIndex]);
        }
      }
      return resultOffersData;
    }
);

export const getCountResultOffers = createSelector(
    getResultOffersData,
    getOffersData,
    (resultOffersData, offersData) => offersData.length ? resultOffersData.length : 0
);

export const getActiveCityName = createSelector(
    getCitiesData,
    getActiveCityIndex,
    (citiesData, activeCityIndex) =>
      citiesData.length ? citiesData[activeCityIndex].name : ``
);

export const getActiveOfferObj = createSelector(
    getOffersData,
    getActiveOfferId,
    (offersData, activeOfferId) => {
      const resObj = offersData.find((offerObj) => offerObj.id === activeOfferId);
      return resObj ? resObj : null;
    }
);

export const getOffersLocationsData = createSelector(
    getResultOffersData,
    getActiveOfferObj,
    (resultOffersData, activeOfferObj) =>
      resultOffersData.map((offerObj) => {
        return {
          location: offerObj.location,
          isActive: offerObj === activeOfferObj,
        };
      })
);

export const getActiveMapLocation = createSelector(
    getCitiesData,
    getActiveCityIndex,
    getActiveOfferObj,
    (citiesData, activeCityIndex, activeOfferObj) => {
      if (!activeOfferObj && citiesData.length) {
        return citiesData[activeCityIndex].location;
      } else if (activeOfferObj) {
        return activeOfferObj.location;
      } else {
        return null;
      }
    }
);

export const getNearData = createSelector(
    getOffersData,
    getActiveOfferObj,
    (offersData, activeOfferObj) => {
      const getDistance = (currLocation) => {
        return Math.sqrt(
            Math.pow(Math.abs(activeOfferObj.location.latitude - currLocation.latitude), 2) +
            Math.pow(Math.abs(activeOfferObj.location.longitude - currLocation.longitude), 2));
      };
      return offersData.map(
          (offerObj) => Object.assign({}, offerObj, {distance: getDistance(offerObj.location)})
      ).sort(
          (a, b) => a.distance > b.distance
      );
    }
);

export const getNearLocationsData = createSelector(
    getNearData,
    getActiveOfferObj,
    (nearData, activeOfferObj) =>
      nearData.map((offerObj) => {
        return {
          location: offerObj.location,
          isActive: offerObj.id === activeOfferObj.id,
        };
      })
);

export const getCityIndexFromOfferId = createSelector(
    getOffersData,
    getCitiesData,
    getActiveOfferId,
    (offersData, citiesData, activeOfferId) => {
      if (offersData.length > 0 && activeOfferId > 0) {
        const cityName = offersData.find((offerObj) => activeOfferId === offerObj.id).city.name;
        return citiesData.findIndex((cityObj) => cityObj.name === cityName);
      } else {
        return 0;
      }
    }
);

export const getResultFavoritesData = createSelector(
    getFavoritesData,
    (favoritesData) => {
      const resultFavoritesData = [];
      favoritesData.forEach((obj) => {
        let index = -1;
        if (resultFavoritesData.length) {
          index = resultFavoritesData.findIndex((resObj) => resObj[0].city.name === obj.city.name);
        }
        if (index === -1) {
          index = resultFavoritesData.push([]) - 1;
        }
        resultFavoritesData[index].push(obj);
      });
      return resultFavoritesData;
    }
);

export const getCountResultFavorites = createSelector(
    getResultFavoritesData,
    (resultFavoritesData) => resultFavoritesData.length
);
