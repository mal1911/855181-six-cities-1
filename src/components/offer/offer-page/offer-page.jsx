import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {offerType} from "../../../prop-types";
import Header from "../../header/index";
import OfferWrapper from "../offer-wrapper/index";
import {connect} from "react-redux";
import {ActionCreator} from "../../../reducer/offers-data/offers-data";
import {getActiveOfferObj} from "../../../reducer/offers-data/selectors";

class OfferPage extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.onDidMountComponent(parseInt(this.props.match.params.id, 10));
  }

  componentWillUnmount() {
  }

  render() {
    const withOfferWrapper = this.props.offerObj ? <OfferWrapper offerObj={this.props.offerObj}/> : null;
    return <React.Fragment>
      <Header/>
      {withOfferWrapper}
    </React.Fragment>;
  }
}

OfferPage.propTypes = {
  match: PropTypes.object,
  offerObj: offerType,
  onDidMountComponent: PropTypes.func,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  offerObj: getActiveOfferObj(state),
});

const mapDispatchToProps = (dispatch) => ({
  onDidMountComponent: (id) => {
    dispatch(ActionCreator.changeActiveOfferId(id));
  },
});

export {OfferPage};

export default connect(mapStateToProps, mapDispatchToProps)(OfferPage);
