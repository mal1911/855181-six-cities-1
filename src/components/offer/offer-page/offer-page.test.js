import React from "react";
import renderer from "react-test-renderer";
import thunk from "redux-thunk";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {OfferPage} from "./offer-page";
import {offersData, initialState} from "../../../mocks/mocks";
import {BrowserRouter} from "react-router-dom";
import {Operation} from '../../../reducer/data/data';

describe(`OfferPage correctly renders`, () => {
  const mockStore = configureStore([thunk]);
  let store;
  let tree;
  jest.mock(`../../../reducer/data/data`);
  Operation.loadCommentsData = () => (dispatch) => dispatch(jest.fn());

  beforeEach(() => {
    store = mockStore(initialState);
    tree = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <OfferPage
              match={{params: {id: `1`}}}
              offerObj={offersData[0]}
              cityIndex={0}
              onLoaded={jest.fn()}
              onChangeActiveOfferObj={jest.fn()}
            />
          </BrowserRouter>
        </Provider>
    ).toJSON();
  });

  it(`OfferPage correctly renders`, () => {
    expect(tree).toMatchSnapshot();
  });
});
