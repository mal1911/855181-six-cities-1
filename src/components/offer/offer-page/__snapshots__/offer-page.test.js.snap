// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferPage correctly renders OfferPage correctly renders 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            aria-current="page"
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                aria-current={null}
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                  style={
                    Object {
                      "backgroundImage": "null",
                    }
                  }
                />
                <span
                  className="header__user-name user__name"
                >
                  Sign In
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>,
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-03.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/room.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
        </div>
      </div>
      ;
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Beautiful & luxurious apartment at great location
            </h1>
            <button
              className="property__bookmark-button button property__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "60%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              3
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Entire 
              room
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              1
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              2
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              231
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Towels
              </li>
              <li
                className="property__inside-item"
              >
                Dishwasher
              </li>
              <li
                className="property__inside-item"
              >
                Breakfast
              </li>
              <li
                className="property__inside-item"
              >
                Washer
              </li>
              <li
                className="property__inside-item"
              >
                Laptop friendly workspace
              </li>
              <li
                className="property__inside-item"
              >
                Air conditioning
              </li>
              <li
                className="property__inside-item"
              >
                Fridge
              </li>
              <li
                className="property__inside-item"
              >
                Baby seat
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="/img/avatar-angelina.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
              <span
                className="property__user-status"
              >
                Pro
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                0
              </span>
            </h2>
            <ul
              className="reviews__list"
            />
          </section>
        </div>
        <section
          className="property__map map"
          id="map"
        />
      </div>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        />
      </section>
    </div>
  </main>,
]
`;
